runtime/model-protobuf/src/main/java/io/aklivity/zilla/runtime/model/protobuf/internal/ProtobufParser.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2021-2024 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.model.protobuf.internal;
+
+import org.antlr.v4.runtime.BailErrorStrategy;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.tree.ParseTreeWalker;
+
+import com.google.protobuf.Descriptors.DescriptorValidationException;
+import com.google.protobuf.Descriptors.FileDescriptor;
+
+import io.aklivity.zilla.runtime.model.protobuf.internal.parser.Protobuf2Lexer;
+import io.aklivity.zilla.runtime.model.protobuf.internal.parser.Protobuf2Parser;
+import io.aklivity.zilla.runtime.model.protobuf.internal.parser.Protobuf3Lexer;
+import io.aklivity.zilla.runtime.model.protobuf.internal.parser.Protobuf3Parser;
+
+public final class ProtobufParser
+{
+    private final FileDescriptor[] dependencies;
+
+    public ProtobufParser(FileDescriptor[] dependencies)
+    {
+        this.dependencies = dependencies != null ? dependencies : new FileDescriptor[0];
+    }
+
+    public FileDescriptor parse(String schemaText)
+    {
+        if (schemaText == null || schemaText.isEmpty())
+        {
+            throw new ProtobufParseException("Schema text cannot be null or empty");
+        }
+
+        // Detect syntax version from schema content
+        ProtobufSyntaxDetector.ProtoSyntax syntax = ProtobufSyntaxDetector.detectSyntax(schemaText);
```

Comment:
Please move the logic to detect syntax into this class as we don't really need the separate concept of a `ProtobufSyntaxDetector`. Instead I think that simplifies to just `ProtobufSyntax` enum.